flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Expressions.java

```diff
@@ -0,0 +1,527 @@
+			Object rightTemporal) {
+		return apiCall(
+			BuiltInFunctionDefinitions.TEMPORAL_OVERLAPS,
+			objectToExpression(leftTimePoint),
+			objectToExpression(leftTemporal),
+			objectToExpression(rightTimePoint),
+			objectToExpression(rightTemporal));
+	}
+
+	/**
+	 * Formats a timestamp as a string using a specified format.
+	 * The format must be compatible with MySQL's date formatting syntax as used by the
+	 * date_parse function.
+	 *
+	 * <p>For example dataFormat('time, "%Y, %d %M") results in strings formatted as "2017, 05 May".
+	 *
+	 * @param timestamp The timestamp to format as string.
+	 * @param format The format of the string.
+	 * @return The formatted timestamp as string.
+	 */
+	public static ApiExpression dateFormat(
+			Object timestamp,
+			Object format) {
+		return apiCall(
+			BuiltInFunctionDefinitions.DATE_FORMAT,
+			objectToExpression(timestamp),
+			objectToExpression(format));
+	}
+
+	/**
+	 * Returns the (signed) number of {@link TimePointUnit} between timePoint1 and timePoint2.
+	 *
+	 * <p>For example, {@code timestampDiff(TimePointUnit.DAY, '2016-06-15'.toDate, '2016-06-18'.toDate} leads
+	 * to 3.
+	 *
+	 * @param timePointUnit The unit to compute diff.
+	 * @param timePoint1 The first point in time.
+	 * @param timePoint2 The second point in time.
+	 * @return The number of intervals as integer value.
+	 */
+	public static ApiExpression timestampDiff(
+			TimePointUnit timePointUnit,
+			Object timePoint1,
+			Object timePoint2) {
+		return apiCall(
+			BuiltInFunctionDefinitions.TIMESTAMP_DIFF,
+			valueLiteral(timePointUnit),
+			objectToExpression(timePoint1),
+			objectToExpression(timePoint2));
+	}
+
+	/**
+	 * Creates an array of literals.
+	 */
+	public static ApiExpression array(Object head, Object... tail) {
+		return apiCall(
+			BuiltInFunctionDefinitions.ARRAY,
+			Stream.concat(
+				Stream.of(head),
+				Stream.of(tail)
+			).map(ApiExpressionUtils::objectToExpression)
+				.toArray(Expression[]::new));
+	}
+
+	/**
+	 * Creates a row of expressions.
+	 */
+	public static ApiExpression row(Object head, Object... tail) {
+		return apiCall(BuiltInFunctionDefinitions.ROW, Stream.concat(
+			Stream.of(head),
+			Stream.of(tail)
+		).map(ApiExpressionUtils::objectToExpression)
+			.toArray(Expression[]::new));
+	}
+
+	/**
+	 * Creates a map of expressions.
+	 */
+	public static ApiExpression map(Object key, Object value, Object... tail) {
+		return apiCall(BuiltInFunctionDefinitions.MAP, Stream.concat(
+			Stream.of(key, value),
+			Stream.of(tail)
+		).map(ApiExpressionUtils::objectToExpression)
+			.toArray(Expression[]::new));
+	}
+
+	/**
+	 * Creates an interval of rows.
+	 *
+	 * @see Table#window(GroupWindow)
+	 * @see Table#window(OverWindow...)
+	 */
+	public static ApiExpression rowInterval(Long rows) {
+		return new ApiExpression(valueLiteral(rows));
+	}
+
+	/**
+	 * Returns a value that is closer than any other value to pi.
+	 */
+	public static ApiExpression pi() {
```

Comment:
I think the reason why it is not a constant is that it is usually a function in other RDBMS:
https://docs.oracle.com/cd/E17952_01/mysql-5.0-en/mathematical-functions.html
https://docs.microsoft.com/en-us/sql/t-sql/functions/pi-transact-sql?view=sql-server-ver15

Whereas the `UNBOUNDED_RANGE` are really symbols/part of the over clause syntax.
As far as I can tell, the intention was to be as close to the sql as possible.

So that we have:
```
table.select(pi()) 
//but
table.window(Over.orderBy($("h")).preceding(CURRENT_RANGE).following(UNBOUNDED_RANGE).as("w")).select(...)
```

The same as in SQL:
```
SELECT PI() FROM table;
//but
SELECT ... OVER (order by h preceding CURRENT_RANGE following UNBOUNDED_RANGE) AS w FROM table
```

Reply:
I'm fine though with changing it to constants as this was also my first thought.